File: app/src/test/java/coolpharaoh/tee/speicher/tea/timer/views/new_tea/suggestions/YellowTeaSuggestionsTest.java
package coolpharaoh.tee.speicher.tea.timer.views.new_tea.suggestions;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.when;

import android.app.Application;
import android.content.res.Resources;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.Mock;
import org.mockito.junit.jupiter.MockitoExtension;

import coolpharaoh.tee.speicher.tea.timer.R;

@ExtendWith(MockitoExtension.class)
class YellowTeaSuggestionsTest {

    private Suggestions yellowTeaSuggestions;

    @Mock
    Application application;
    @Mock
    Resources resources;

    @BeforeEach
    void setUp() {
        when(application.getResources()).thenReturn(resources);
        yellowTeaSuggestions = new YellowTeaSuggestions(application);
    }

    @Test
    void getAmountTsSuggestion() {
        final int[] arrayTs = new int[]{1, 2};
        when(resources.getIntArray(R.array.new_tea_suggestions_yellow_tea_amount_ts)).thenReturn(arrayTs);

        assertThat(yellowTeaSuggestions.getAmountTsSuggestions()).isEqualTo(arrayTs);
    }

    @Test
    void getAmountGrSuggestion() {
        final int[] arrayGr = new int[]{1, 2};
        when(resources.getIntArray(R.array.new_tea_suggestions_yellow_tea_amount_gr)).thenReturn(arrayGr);

        assertThat(yellowTeaSuggestions.getAmountGrSuggestions()).isEqualTo(arrayGr);
    }

    @Test
    void getAmountTbSuggestion() {
        final int[] arrayTb = new int[]{1, 2};
        when(resources.getIntArray(R.array.new_tea_suggestions_yellow_tea_amount_tb)).thenReturn(arrayTb);

        assertThat(yellowTeaSuggestions.getAmountTbSuggestions()).isEqualTo(arrayTb);
    }

    @Test
    void getTemperatureCelsiusSuggestion() {
        final int[] arrayCelsius = new int[]{1, 2};
        when(resources.getIntArray(R.array.new_tea_suggestions_yellow_tea_temperature_celsius)).thenReturn(arrayCelsius);

        assertThat(yellowTeaSuggestions.getTemperatureCelsiusSuggestions()).isEqualTo(arrayCelsius);
    }

    @Test
    void getTemperatureFahrenheitSuggestion() {
        final int[] arrayFahrenheit = new int[]{1, 2};
        when(resources.getIntArray(R.array.new_tea_suggestions_yellow_tea_temperature_fahrenheit)).thenReturn(arrayFahrenheit);

        assertThat(yellowTeaSuggestions.getTemperatureFahrenheitSuggestions()).isEqualTo(arrayFahrenheit);
    }

    @Test
    void getSteepingTimeSuggestion() {
        final String[] arrayTime = new String[]{"1:00", "2:30"};
        when(resources.getStringArray(R.array.new_tea_suggestions_yellow_tea_time)).thenReturn(arrayTime);

        assertThat(yellowTeaSuggestions.getTimeSuggestions()).isEqualTo(arrayTime);
    }
}

File: app/src/test/java/coolpharaoh/tee/speicher/tea/timer/views/new_tea/suggestions/YellowTeaSuggestionsTest.kt
package coolpharaoh.tee.speicher.tea.timer.views.new_tea.suggestions

import android.app.Application
import android.content.res.Resources
import coolpharaoh.tee.speicher.tea.timer.R
import io.mockk.every
import io.mockk.impl.annotations.InjectMockKs
import io.mockk.impl.annotations.MockK
import io.mockk.junit5.MockKExtension
import org.assertj.core.api.Assertions.*
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith

@ExtendWith(MockKExtension::class)
internal class YellowTeaSuggestionsTest {
    @MockK
    lateinit var application: Application

    @MockK
    lateinit var resources: Resources

    @InjectMockKs
    lateinit var yellowTeaSuggestions: YellowTeaSuggestions

    @BeforeEach
    fun setUp() {
         every { application.resources } returns resources
    }

    @Test
    fun getAmountTsSuggestion() {
        val arrayTs = intArrayOf(1, 2)
         every { resources.getIntArray(R.array.new_tea_suggestions_yellow_tea_amount_ts) } returns arrayTs

        assertThat(yellowTeaSuggestions.amountTsSuggestions).isEqualTo(arrayTs)
    }

    @Test
    fun getAmountGrSuggestion() {
        val arrayGr = intArrayOf(1, 2)
         every { resources.getIntArray(R.array.new_tea_suggestions_yellow_tea_amount_gr) } returns arrayGr

        assertThat(yellowTeaSuggestions.amountGrSuggestions).isEqualTo(arrayGr)
    }

    @Test
    fun getAmountTbSuggestion() {
        val arrayTb = intArrayOf(1, 2)
         every { resources.getIntArray(R.array.new_tea_suggestions_yellow_tea_amount_tb) } returns arrayTb

        assertThat(yellowTeaSuggestions.amountTbSuggestions).isEqualTo(arrayTb)
    }

    @Test
    fun getTemperatureCelsiusSuggestion() {
        val arrayCelsius = intArrayOf(1, 2)
         every { resources.getIntArray(R.array.new_tea_suggestions_yellow_tea_temperature_celsius) } returns arrayCelsius

        assertThat(yellowTeaSuggestions.temperatureCelsiusSuggestions).isEqualTo(arrayCelsius)
    }

    @Test
    fun getTemperatureFahrenheitSuggestion() {
        val arrayFahrenheit = intArrayOf(1, 2)
         every { resources.getIntArray(R.array.new_tea_suggestions_yellow_tea_temperature_fahrenheit) } returns arrayFahrenheit

        assertThat(yellowTeaSuggestions.temperatureFahrenheitSuggestions).isEqualTo(arrayFahrenheit)
    }

    @Test
    fun getSteepingTimeSuggestion() {
        val arrayTime = arrayOf("1:00", "2:30")
         every { resources.getStringArray(R.array.new_tea_suggestions_yellow_tea_time) } returns arrayTime

        assertThat(yellowTeaSuggestions.timeSuggestions).isEqualTo(arrayTime)
    }
}